### DISCUSSION TITLE
Which n_neighbors shall we optimize in scVelo?

### DISCUSSION THREAD

**Discussion by hyjforesight** [theislab/scvelo#754] [Q&A] [answered]
Hello scVelo,
In Velocyto paper, the authors mentioned the limitation of RNA velocity: "The neighborhood-based procedure has several biases and unintuitive effects. First, if the neighborhood size is larger than the distance to the ultimate trajectory target, it is possible to have edge effects that will rotate or even invert the direction of the projection away from its intended target". It means the number of neighbors should be optimized. 
I'm wondering in which function shall we optimized n_neighbors, scv.pp.moments(), scv.tl.velocity_graph() or scv.pl.velocity_embedding_stream()? Shall the n_neigbors be the same among these 3 functions?
Thanks!
Best,
YJ

**Comment by WeilerP** [comment, marked as answer]
@hyjforesight, in the scvelo paper, it was shown that results are robust w.r.t. the exact number of neighbors chosen for imputation. This statement is, of course, only true when working in a reasonable range. You should choose the number of neighbors once when calculating moments. Afterwards, the number of neighbors is kept fixed.
